cli/report/report.py

```diff
@@ -269,18 +270,110 @@ def file_jira_issues(self) -> list[str]:
                 step_name=pair["failure"]["step"],
             )
-
-            jira_issue = self.jira.create_issue(
+            labels = [
+                self.job_name,
+                pair["failure"]["step"],
+                pair["rule"]["classification"],
+                pair["failure"]["failure_type"],
+            ]
+
+            # Find duplicate bugs
+            duplicate_bugs = self.find_duplicate_bugs(
                 project=project,
-                summary=summary,
-                description=description,
-                issue_type=type,
-                file_attachments=file_attachments,
+                job_name=self.job_name,
+                failed_step=pair["failure"]["step"],
+                failure_type=pair["failure"]["failure_type"],
             )
-            bugs_filed.append(jira_issue.key)
+
+            # If duplicates are found, comment
+            if len(duplicate_bugs) > 0:
+                for bug in duplicate_bugs:
+                    self.add_duplicate_comment(
+                        bug_key=bug,
+                        failed_step=pair["failure"]["step"],
+                        classification=pair["rule"]["classification"],
+                    )
+
+            # If duplicates are not found, file a bug
+            else:
+                jira_issue = self.jira.create_issue(
+                    project=project,
+                    summary=summary,
+                    description=description,
+                    issue_type=type,
+                    file_attachments=file_attachments,
+                    labels=labels,
+                )
+                bugs_filed.append(jira_issue.key)
 
         return bugs_filed
 
+    def find_duplicate_bugs(
+        self,
+        project: str,
+        job_name: Optional[str],
+        failed_step: str,
+        failure_type: str,
+    ) -> list[str]:
+        """
+        Searches Jira for possible duplicate bugs in the given project. If a bug in the same job, with the same failure_type, and the same failed_step, is found a list of duplicate bugs is returned.
+
+        :param project: Jira project to search in
+        :param job_name: Job name to search for
+        :param failed_step: Failed step name to search for
+        :param failure_type: Failure type to match
+
+        :return: A list of duplicate bugs that are still open
+        """
+
+        self.logger.info(
+            f'Searching for duplicate bugs in project {project} for a "{failure_type}" failure type in the {failed_step} step.',
+        )
+
+        # This JQL query will find any bug in the provided project that:
+        # Has a label that matches the job name
+        # AND has a label that matches the failed step name
+        # AND has a label that matches the failure type
+        # AND the bugs are not closed
+        jql_query = f'project = {project} AND labels="{job_name}" AND labels="{failed_step}" AND labels="{failure_type}" AND status not in (closed)'
```

Comment:
1. The optional thing for job name is forced by MyPy for some reason. The job name will always be set, though. It takes the value from the command line arguments or from the `$JOB_NAME` environment variable
2. I think we should just file a new one because the same type of failure could have happened in the past for a different reason and it probably needs to be investigated again.